### ISSUE TITLE
FNX-4539 ⁃ [Bug]Native code crash on Beta 5.1.0-beta.1 after quickly opening one by one the featured Top Sites

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#10766]
## Steps to reproduce
1. Open Pocket from the Top Sites section
1. Tap the tab counter
1. Open Wikipedia from the Top Sites section
1. Tap the tab counter
1. Open You Tube  from the Top Sites section
1. Tap the  ⋮ Main menu

### Expected behavior
Properly open the websites and the  ⋮ Main menu

### Actual behavior
Native Code Crash

### Device information
* Android device: OnePlus A3 (Android 6.0.1)
* Fenix version: **`Beta 5.1.0-beta.1 from 5/19`**

### Notes
❗ Not reproducible 100% of the time
❗ Didn't manage to reproduce on the latest **`Nightly Build 200519 from 5/19`**

► Video
![20200519-144948](https://user-images.githubusercontent.com/51314259/82324708-be255e00-99e2-11ea-86e6-fab97fb2c8c3.gif)

► Youtube message
![rsz](https://user-images.githubusercontent.com/51314259/82325215-80750500-99e3-11ea-9e35-121c3371d175.png)UNITO-UNDERSCORE!screenshot!UNITO-UNDERSCORE!20200519-145200!

```kotlin
05-19 14:53:40.189 1302-2464/? I/ActivityManager: Start proc 29385:org.mozilla.fenix.beta:mozilla.components.lib.crash.CrashHandler/u0a283 for service org.mozilla.fenix.beta/mozilla.components.lib.crash.handler.CrashHandlerService
05-19 14:53:40.397 28975-28975/? W/GeckoSession: Attempted to close a GeckoSession that was already closed.
05-19 14:53:40.399 28975-28975/? I/art: Note: end time exceeds epoch: 
05-19 14:53:40.402 28975-29115/? I/Gecko: [GFX1-]: Failed to create EGLContext!: 0x300c
05-19 14:53:40.420 29385-29398/? E/mozac/CrashReporter: CrashHandlerService received native code crash
05-19 14:53:40.423 29385-29398/? I/mozac/CrashReporter: Received crash: NativeCodeCrash(minidumpPath=/data/user/0/org.mozilla.fenix.beta/files/mozilla/Crash Reports/pending/5552c697-25b5-fa8c-2d3b-ee120e3bbb23.dmp, minidumpSuccess=false, extrasPath=/data/user/0/org.mozilla.fenix.beta/files/mozilla/Crash Reports/pending/5552c697-25b5-fa8c-2d3b-ee120e3bbb23.extra, isFatal=false, breadcrumbs=[], uuid=2c2520b7-473c-4041-9cd1-472475c1101d)
05-19 14:53:40.424 28975-28975/? W/GeckoSession: Attempted to close a GeckoSession that was already closed.
05-19 14:53:40.425 28975-28975/? I/art: Note: end time exceeds epoch: 
05-19 14:53:40.428 28975-28975/? W/GeckoSession: Attempted to close a GeckoSession that was already closed.
05-19 14:53:40.428 28975-28975/? I/art: Note: end time exceeds epoch: 
05-19 14:53:40.429 28975-29115/? I/Gecko: [GFX1-]: Failed to create EGLContext!: 0x300c
05-19 14:53:40.456 28975-28975/? W/GeckoEventDispatcher: No listener for GeckoView:AccessibilityEnabled
05-19 14:53:40.461 28975-29115/? I/Gecko: [GFX1-]: Failed to create EGLContext!: 0x300c
05-19 14:53:40.495 1302-1302/? I/DeviceIdleController: updateChargingLocked: charging=true
05-19 14:53:40.495 1302-1302/? I/DeviceIdleController: becomeActiveLocked, reason = charging
05-19 14:53:40.513 1302-2209/? I/ActivityManager: Start proc 29400:org.mozilla.fenix.beta:tab0/u0a283 for service org.mozilla.fenix.beta/org.mozilla.gecko.process.GeckoChildProcessServices$tab0
05-19 14:53:40.515 29385-29398/? I/mozac/CrashReporter: Invoking non-fatal PendingIntent
05-19 14:53:40.516 1302-2464/? I/ActivityManager: START u0 {flg=0x14000000 cmp=org.mozilla.fenix.beta/org.mozilla.fenix.HomeActivity (has extras)} from uid 10283 pid -1 on display 0
05-19 14:53:40.518 28975-29054/? I/mozac/CrashReporter: Crash report submitted to 1 telemetry services
05-19 14:53:40.519 29385-29398/? I/art: System.exit called, status: 0
05-19 14:53:40.519 29385-29398/? I/AndroidRuntime: VM exiting with result code 0, cleanup skipped.
05-19 14:53:40.519 28975-29016/? D/ServiceAllocator: org.mozilla.gecko.process.GeckoChildProcessServices$tab0 updateBindings: BACKGROUND priority, 0 importance, 2 successful binds, 0 failed binds, 0 successful unbinds, 0 failed unbinds
05-19 14:53:40.567 1302-1729/? D/ConnectivityService: releasing NetworkRequest NetworkRequest [ id=175, legacyType=-1, [ Transports: WIFI Capabilities: NOT_RESTRICTED&TRUSTED&NOT_VPN] ]
05-19 14:53:40.571 1302-2216/? I/ActivityManager: Process org.mozilla.fenix.beta:mozilla.components.lib.crash.CrashHandler (pid 29385) has died
05-19 14:53:40.572 1302-2216/? W/ActivityManager: Scheduling restart of crashed service org.mozilla.fenix.beta/mozilla.components.lib.crash.handler.CrashHandlerService in 1000ms
05-19 14:53:40.574 1302-1342/? V/ActivityManager: killProcessGroupAsync took 0 ms for PID 29385 on thread 14
05-19 14:53:40.602 28975-29041/? W/FirefoxAccountStateMach: Got invalid event Event$Init for state NotAuthenticated.
05-19 14:53:40.607 28975-28975/? D/PhoneWindow: state.preset = true, color: 0, translucent = 0
05-19 14:53:40.742 29400-29400/? D/GeckoThread: State changed to LAUNCHED
05-19 14:53:40.744 29400-29414/? I/GeckoThread: preparing to run Gecko
05-19 14:53:40.752 28975-29016/? D/ServiceAllocator: org.mozilla.gecko.process.GeckoChildProcessServices$tab0 updateBindings: FOREGROUND priority, 2 importance, 1 successful binds, 0 failed binds, 0 successful unbinds, 0 failed unbinds
05-19 14:53:40.759 29400-29414/? D/GeckoThread: State changed to MOZGLUE_READY```

**Comment by pocmo** [comment]
Do you happen to have a crash-stats link for this native code crash? :)

**Comment by liuche** [comment]
I went through the (few) crashes from 5.1.0-beta.1 to try to find another OnePlus crasher, but the only signatures I could find with OnePlus were from #10770 - maybe it's the same root cause?

**Comment by AndiAJ** [comment]
Hi @liuche & @pocmo , I've just managed to repro the issue on 5.1.0-beta.1.
It seems that 2 reports got generated:
• https://crash-stats.mozilla.org/report/index/7a219944-5fc1-48fd-bb11-4b8190200520
• https://crash-stats.mozilla.org/report/index/38b3fc31-6dda-4159-8070-641e10200520

![image](https://user-images.githubusercontent.com/51314259/82423485-7ce98880-9a8c-11ea-8042-88052ab867de.png)

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on **`Beta 5.1.0-beta.2 from 5/27`** using a OnePlus A3 (Android 6.0.1)

► Video
![20200527-160559](https://user-images.githubusercontent.com/51314259/83022775-b3974400-a034-11ea-8305-1cea5878f5ea.gif)
